client/mapHandler.cpp

```diff
@@ -567,7 +567,7 @@ void CMapHandler::CMapWorldViewBlitter::drawOverlayEx(SDL_Surface * targetSurf)
 			continue;
 
 		realPos.x = initPos.x + (iconInfo.pos.x - topTile.x) * tileSize;
-		realPos.y = initPos.x + (iconInfo.pos.y - topTile.y) * tileSize;

```

Comment:
And this is really interesting)

Reply:
I believe the right fix is to use `initPos.y` on this line